### ISSUE TITLE
error when password contains %

### ISSUE THREAD

**Issue by karfau** [dboissier/mongo4idea#145]
the following error occurs in the case that the password contains (in my case starts with) `%`:
so lets assume my password would be `%abc123`
```
URLDecoder: Illegal hex characters in escape (%) pattern - For input string: "ab"
java.lang.IllegalArgumentException: URLDecoder: Illegal hex characters in escape (%) pattern - For input string: "ab"
	at java.net.URLDecoder.decode(URLDecoder.java:192)
	at com.mongodb.ConnectionString.<init>(ConnectionString.java:243)
	at com.mongodb.MongoClientURI.<init>(MongoClientURI.java:163)
	at com.mongodb.MongoClientURI.<init>(MongoClientURI.java:150)
	at org.codinjutsu.tools.mongo.logic.MongoManager.createMongoClient(MongoManager.java:302)
	at org.codinjutsu.tools.mongo.logic.MongoManager.loadDatabaseCollections(MongoManager.java:92)
	at org.codinjutsu.tools.mongo.logic.MongoManager.loadServer(MongoManager.java:81)
	at org.codinjutsu.tools.mongo.view.MongoExplorerPanel$2.run(MongoExplorerPanel.java:134)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:366)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:55)
```
This happens independent on the AuthMechanism or wether I use SSL connection or not.

I already tried to enter `%25abc123` into the password field, as to escape the `%` sign,
but then it fails because of wrong password, of course...

Any idea?

I'm using your Plugin inside PyCharm Community Edition 5.0.4, Build number PC-143.1919 unter Ubuntu 14.4, just installed your plugin so Version is 0.7.3

**Comment by dboissier** [comment]
Hi,

The current code build a mongoURI and the % character is not supported. I will fix it ASAP.

Thanks,

David

**Comment by dboissier** [comment]
Hi,

Does the latest version of the plugin work on your IDE?

**Comment by karfau** [comment]
I will check as soon as I'm back at the office, but this will take more then a week from now.
(I worked around the problem connecting to the db with another user, so I didn't check again. ;) )

**Comment by karfau** [comment]
Actually currently I'm not able to connect using a mongo config that worked before and which works fine on the command line and this doesn't depend on the % sign in the password. 
Also a configuration to localhost that i have works fine

The error is

    source can not be null
    java.lang.IllegalArgumentException: source can not be null
	at com.mongodb.assertions.Assertions.notNull(Assertions.java:37)
	at com.mongodb.MongoCredential.<init>(MongoCredential.java:233)
	at com.mongodb.MongoCredential.createPlainCredential(MongoCredential.java:174)
	at org.codinjutsu.tools.mongo.logic.MongoManager.getMongoCredential(MongoManager.java:309)
	at org.codinjutsu.tools.mongo.logic.MongoManager.createMongoClient(MongoManager.java:301)
	at org.codinjutsu.tools.mongo.logic.MongoManager.loadDatabaseCollections(MongoManager.java:92)
	at org.codinjutsu.tools.mongo.logic.MongoManager.loadServer(MongoManager.java:81)
	at org.codinjutsu.tools.mongo.view.MongoExplorerPanel$3.run(MongoExplorerPanel.java:153)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

but from the fields in the config dialog nothing looks strange.
Should we create a new ticket for it?
Beside that any ideas, anything to try out?

**Comment by dboissier** [comment]
Password issue is fixed. The bug you mention has been fixed in the latest build 0.8.0 (Idea 15) / 0.8.1 (Idea 2016)
